src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1680,6 +1711,55 @@ void GrpcLb::CreateOrUpdateChildPolicyLocked() {
   child_policy_->UpdateLocked(std::move(update_args));
 }
 
+//
+// subchannel caching
+//
+
+void GrpcLb::CacheDeletedSubchannel(
+    RefCountedPtr<SubchannelInterface> subchannel) {
+  grpc_millis deletion_time =
+      ExecCtx::Get()->Now() + subchannel_cache_interval_ms_;
+  cached_subchannels_[deletion_time].push_back(std::move(subchannel));
+  if (!subchannel_cache_timer_pending_) {
+    Ref(DEBUG_LOCATION, "OnSubchannelCacheTimer").release();
+    subchannel_cache_timer_pending_ = true;
+    StartSubchannelCacheTimer();
+  }
+}
+
+void GrpcLb::StartSubchannelCacheTimer() {
+  GPR_ASSERT(!cached_subchannels_.empty());
+  grpc_timer_init(&subchannel_cache_timer_, cached_subchannels_.begin()->first,
+                  &on_subchannel_cache_timer_);
+}
+
+void GrpcLb::OnSubchannelCacheTimer(void* arg, grpc_error_handle error) {
+  auto* self = static_cast<GrpcLb*>(arg);
+  GRPC_ERROR_REF(error);
+  self->work_serializer()->Run(
+      [self, error]() { self->GrpcLb::OnSubchannelCacheTimerLocked(error); },
+      DEBUG_LOCATION);
+}
+
+void GrpcLb::OnSubchannelCacheTimerLocked(grpc_error_handle error) {
+  if (subchannel_cache_timer_pending_ && error == GRPC_ERROR_NONE) {
```

Comment:
looks like we need to reset `subchannel_cache_timer_pending_` in here?

Reply:
Good catch.  Done.